### PR TITLE
Rework the changelog

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Review by mseri — APPROVED** [review]

**Review by mseri — APPROVED** [review]

File: CHANGES.md [hunk 1]
```diff
@@ -1,66 +1,16 @@
-## current
+## v4.0.0 (2021-03-16)
```
Comment (mseri) [inline CHANGES.md:1]: Should we keep this as current until we have a story for the new headers module?

Comment (samoht) [reply CHANGES.md]: fixed

File: CHANGES.md [hunk 2]
```diff
@@ -71,6 +21,12 @@
 - cohttp-lwt: partly inline read_response, fix body stream leak (@madroach @dinosaure #696)
 - improve media type parsing (@seliopou #542, @dinosaure #725)
 - add comparison functions for Request.t and Response.t via ppx_compare (@msaffer-js @dinosaure #686)
+- [reverted] breaking changes to client and server API to use conduit
+  3.0.0 (@dinosaure #692). However, as the design discussion did not reach
+  consensus, these changes were reverted to preserve better compatibility with
+  existing cohttp users. (#741,  @samoht)
+
+## v3.0.0 - aborted
```
Comment (mseri) [inline CHANGES.md:29]: Sorry to bother again.Can you rebase on master? I don’t understand why it is not showing up the changes from the marged PRs

Comment (samoht) [reply CHANGES.md:30]: done

**Comment by mseri** [comment]
Thanks!

**Pull request merged**